.github/workflows/build_and_test.yml

```diff
@@ -122,9 +122,11 @@ jobs:
 
   check-codestyle-rust:
     needs: [ check-permissions, build-build-tools-image ]
+    # There's no reason to expect clippy or code formatting to be different on different platforms,
+    # so it's enough to run these on x64 only.
     strategy:
       matrix:
-        arch: [ x64, arm64 ]
```

Comment:
Initially, it was added within https://github.com/neondatabase/neon/issues/5829. Back then, we didn't build and run tests on ARM.

I assume `clippy` expands macros only for the current os/arch, please correct me if I'm wrong.

If I'm right, it still useful to catch issues with the `cfg` macro and `target_arch` being used, such as unused variables/functions.
So far, we have it only in one place (in tests):
```
$ git grep target_arch
pageserver/src/tenant/storage_layer/layer/tests.rs:#[cfg(target_arch = "x86_64")]
```

But some more might come from dependencies, I guess.

Reply:
I'm totally Ok with losing clippy coverage of that test, if that makes the CI faster.

cc @problame since you authored #5829

Reply:
I've noticed that we use `small` runners for the job, let me try switching it to `large` and see if it's still that bad

Reply:
Using `large` runners didn't  help at all

Reply:
Conrad suggested using `--partition` for `cargo hack` and run it in parallel, going to try this

Reply:
Using `--partitions` helped! (tried 8 so far)

- x64: from 15m to 7m
- arm: from 40m to 17m